### PR TITLE
[wip] improved github actions workflow

### PR DESCRIPTION
**Pull request by stas00**
This is a WIP work on https://github.com/huggingface/transformers/issues/7887

### CODE REVIEW

File: .github/workflows/experiment.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+name: Self-hosted runner (scheduled)
+
+on:
+  push:
+    branches:
+      - ci_*
```
Comment (sshleifer) [inline .github/workflows/experiment.yml:6]: I think this needs to be modified to `stas00:*` or something that matches your branch.

**Comment by stas00** [comment]
@sshleifer, I don't think I can do anything here w/o repo permissions. I have no access to the runners.

File: .github/workflows/experiment.yml (hunk 1 above)
Comment (stas00) [reply .github/workflows/experiment.yml]: there is no `username` here, just the branch name. 

To fix the integration job, I had to change `github-torch-hub.yml` for it to use my username when it tries to download a zip of the branch code - and then this job finally works, but it is of no use to what we are trying here.

**Comment by stas00** [comment]
github is so borked! It now won't let me delete the new workflow - and keeps running it automatically and emailing me on each commit I make elsewhere in this project - what horrors!

**Comment by sshleifer** [comment]
ask on community.github.com, they are pretty responsive.

**Pull request closed without merging**